server/src/main/scala/org/http4s/server/package.scala

```diff
@@ -30,43 +59,19 @@ package object server {
       *
       */
     def const[A, B](b: => Task[B]): Service[A, B] = b.liftKleisli
-  }
 
-  /**
-   * A [[Service]] that returns an optional result.
-   */
-  type PartialService[A, B] = Kleisli[({type L[x] = OptionT[Task, x]})#L, A, B]
-
-  object PartialService {
     /**
-     * Lifts an unwrapped function that returns an OptionT over Task into a [[PartialService]].
-     */
-    def lift[A, B](f: A => OptionT[Task, B]): PartialService[A, B] =
-      Kleisli.kleisliU(f)
-
-    /**
-      *  Lifts (by sequencing) an unwrapped function that returns an optional
-      *  Task into a [[PartialService]].
+      *  Lifts a value into a [[Service]].
+      *
       */
-    def liftS[A, B](f: A => Option[Task[B]]): PartialService[A, B] =
-      lift(f.andThen(o => OptionT(o.sequence)))
+    def constVal[A, B](b: => B): Service[A, B] = Task.now(b).liftKleisli
 
     /**
-     * Lifts a partial function that returns a Task into a [[PartialService]].  Where the
-     * function is not defined, the [[PartialService]] returns `OptionT.none`
-     */
-    def liftPF[A, B](pf: PartialFunction[A, Task[B]]): PartialService[A, B] =
-      liftS(pf.lift)
-  }
-
-  implicit class PartialServiceSyntax[A, B](val service: PartialService[A, B]) extends AnyVal {
-    def or(default: => Task[B]): Service[A, B] =
-      service.mapK(_.getOrElseF(default))
-
-    def orElse(that: PartialService[A, B]): PartialService[A, B] =
-      PartialService.lift { req =>
-        service.run(req).orElse(that.run(req))
-      }
+      * Allows Service chainig through an implicit [[Fallthrough]] instance.
+      *
+      */
+    def orElse[A, B : Fallthrough](a: Service[A, B], b: Service[A, B]): Service[A, B] =
```

Comment:
I had imagined it taking a middleware-like syntax where the last param list was the Service to modify:

```scala
def withFallback[A, B: Fallthrough](fallback: Service[A, B])(service: Service[A, B]): Service[A, B]
```

For the infix syntax we all know and love:

```scala
implicit class ServiceSyntax[A, B](service: Service[A, B]) {
  def orElse(fallback: Service[A, B]) = withFallback(fallback)(service)
}
```